### DISCUSSION TITLE
getValues('field') vs getValues().field

### DISCUSSION THREAD

**Discussion by terebentina** [react-hook-form/react-hook-form#3345] [Q&A] [answered]
Sorry for spamming the discussions, I am trying to get my head around the quirks of react-hook-form atm.
Is there a difference between getValues('field') and getValues().field? The documentation seems to say you can use either.
In my case the former returns undefined while the latter returns the proper data for an array field:
Watch the console in this CSB and click on the first button then on the second:
https://codesandbox.io/s/interesting-hoover-p0qfn?file=/src/App.js

**Comment by bluebill1049** [comment, marked as answer]
getValues() return everything from the form 
getValues('name') return speicifc field

**Reply by terebentina** [reply]
Yes, but I am asking why does `getValues().field` return my field but `getValues('field')` returns undefined for the same field.
To me it looks like a bug, that's why I created the CSB. Could you please have a look and click on the 2 buttons?

**Reply by bluebill1049** [reply]
that's not a field name -> methods.getValues("blocks") rather than a `key`

**Reply by terebentina** [reply]
ahh, gotcha, thanks!

**Comment by bluebill1049** [comment]
<img width="1068" alt="Screen Shot 2020-11-04 at 7 31 19 pm" src="https://user-images.githubusercontent.com/10513364/98087518-52717780-1ed4-11eb-9e0d-40c696bdd36a.png">
